tool/speed.cc

```diff
@@ -1314,6 +1314,24 @@ static bool Speed25519(const std::string &selected) {
 
   results.Print("Curve25519 arbitrary point multiplication");
 
+  if (!TimeFunction(&results, []() -> bool {
+        uint8_t out_base[32], in_base[32];
+        BM_memset(in_base, 0, sizeof(in_base));
```

Comment:
Is there a difference between this and `uint8_t in_base[32] = {0}` for performance?

Reply:
Not really.

Existing:
```
Did 18000 X25519 (base-point + arbitrary point multiplication) operations in 1055209us (17058.2 ops/sec)
```

Initialise with `0`:
```
Did 18000 X25519 (base-point + arbitrary point multiplication) operations in 1054890us (17063.4 ops/sec)
```

Diff:
```
diff --git a/tool/speed.cc b/tool/speed.cc
index 1f8947e2b..3c951351a 100644
--- a/tool/speed.cc
+++ b/tool/speed.cc
@@ -1315,13 +1315,13 @@ static bool Speed25519(const std::string &selected) {
   results.Print("Curve25519 arbitrary point multiplication");
 
   if (!TimeFunction(&results, []() -> bool {
-        uint8_t out_base[32], in_base[32];
-        BM_memset(in_base, 0, sizeof(in_base));
+        uint8_t out_base[32], in_base[32] = {0};
+        //BM_memset(in_base, 0, sizeof(in_base));
         X25519_public_from_private(out_base, in_base);
 
-        uint8_t out[32], in1[32], in2[32];
-        BM_memset(in1, 0, sizeof(in1));
-        BM_memset(in2, 0, sizeof(in2));
+        uint8_t out[32], in1[32] = {0}, in2[32] = {0};
+        //BM_memset(in1, 0, sizeof(in1));
+        //BM_memset(in2, 0, sizeof(in2));
         in1[0] = 1;
         in2[0] = 9;
         return X25519(out, in1, in2) == 1;
```